src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -93,6 +96,50 @@
   (instance? X509CRL x))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ValidX500Name
+  (schema/pred valid-x500-name?))
+
+(def Extension
+  "A map containing all the fields required to define an extension."
+  {:oid schema/Str
+   :critical (schema/pred #(not (nil? %)))
+   :value (schema/pred #(not (nil? %)))})
+
+(def ExtensionList
+  [Extension])
+
+(def CertOrCSR
+  (schema/cond-pre X509Certificate PKCS10CertificationRequest))
+
+(def PublicOrPrivateKey
+  (schema/cond-pre PublicKey PrivateKey))
+
+(def ExtensionContainer
+  (schema/cond-pre
+   X509Certificate
+   PKCS10CertificationRequest
+   X509CRL
+   List))
+
+(def SSLContextOptions
+  {(schema/optional-key :ssl-cert) schema/Any
+   (schema/optional-key :ssl-key) schema/Any
+   (schema/optional-key :ssl-ca-cert) schema/Any
+   (schema/optional-key :ssl-ca-crls) schema/Any
+   (schema/optional-key :ssl-context) SSLContext})
+
+(def Something
```

Comment:
I'd prefer `Object` - and also in the value for the `:value` key in `Extension` above.